planty/posts/views.py

```diff
@@ -45,5 +51,100 @@ class UpdatePostView(LoginRequiredMixin, UserIsAuthorMixin, UpdateView):
 class DeletePostView(LoginRequiredMixin, UserIsAuthorMixin, DeleteView):
     model = Post
     # Needs a url to redirect after deleted
-    # TODO change url to explore page
-    success_url = reverse_lazy('login')
+    success_url = reverse_lazy('explore')
+
+
+def explore(request):
+    if request.method == 'POST':
+        return HttpResponseBadRequest()
+
+    # Categories
+    user_categories = request.GET.getlist('category')
+
+    if len(user_categories) == 0:  # No category checkbox is selected
+        posts_to_display = Post.objects.all()
+
+    else:
+        if 'ALL' in user_categories:
+            posts_to_display = Post.objects.all()
+        else:
+            if 'O' in user_categories:
+                cat1 = 'O'
+            else:
+                cat1 = None
+
+            if 'I' in user_categories:
+                cat2 = 'I'
+            else:
+                cat2 = None
+
+            if 'F' in user_categories:
+                cat3 = 'F'
+            else:
+                cat3 = None
+
+            posts_to_display = Post.objects.filter(Q(plant_category=cat1)
+                                                   | Q(plant_category=cat2)
+                                                   | Q(plant_category=cat3))
+
+    # Price range
+    max_aggregated_value = Post.objects.all().aggregate(Max('price'))
+    max_price = max_aggregated_value['price__max']
+    min_aggregated_value = Post.objects.all().aggregate(Min('price'))
+    min_price = min_aggregated_value['price__min']
+
+    user_max_price = request.GET.get('price_value', None)
+
+    if user_max_price is not None:
+        posts_to_display = posts_to_display.filter(price__lte=user_max_price)
+
+    # Basic Search
+    searched = request.GET.get('searched', None)
+    if searched is not None and searched != "":
+        posts_to_display = posts_to_display.filter(title__icontains=searched)
+
+    # Date Posted (doesn't work for different timezones)
+    last_day = datetime.now() - timedelta(days=1)
+    last_week = datetime.now() - timedelta(days=7)
+    last_month = datetime.now() - timedelta(days=30)
+
+    user_date = request.GET.get('date', None)
+
+    if user_date is not None:
+        if user_date == "day":
+            date_filter = last_day
+        elif user_date == "week":
+            date_filter = last_week
+        else:
+            date_filter = last_month
+
+        posts_to_display = posts_to_display.filter(
+            date_posted__gte=date_filter)
+
+    # Sort Results
+    sort_by = request.GET.get('sort-by', None)
+
+    if sort_by is None or sort_by == "newer":
+        posts = posts_to_display.order_by('-date_posted')
+    elif sort_by == "older":
+        posts = posts_to_display.order_by('date_posted')
+    elif sort_by == "higher":
+        posts = posts_to_display.order_by('-price')
+    else:
+        posts = posts_to_display.order_by('price')
+
+    # Pagination
+    post_list = posts
```

Comment:
No need to create a new variable `post_list` if you only use it in the next line. Simply do `paginator = Paginator(posts, 12)`